### PR TITLE
Project details

### PR DESCRIPTION
**Pull request by GSuma21**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,6 +24,7 @@
     "@ngx-translate/core": "^15.0.0",
     "@ngx-translate/http-loader": "^8.0.0",
     "authentication_frontend_library": "^0.0.18",
+    "lodash": "^4.17.21",
```
Comment (ramkumar-ns-dev) [inline package.json]: why do we need lodash ?

File: projects/lib-project/src/lib/dock/components/project-details/project-details.component.ts [hunk 2]
```diff
@@ -1,27 +1,40 @@
...
+  constructor(private activatedRoute: ActivatedRoute) {
   }
 
-  onButtonClick(buttonTitle: string) {
+  ngOnInit(){
+    this.state$ = this.activatedRoute.paramMap
+    .pipe(map(() => window.history.state));
+
+  // Subscribe to the state$ observable to log the state to the console
+  this.stateSubscription = this.state$.subscribe(state => {
```
Comment (ramkumar-ns-dev) [inline projects/lib-project/src/lib/dock/components/project-details/project-details.component.ts]: why it's not unsubscribed ?

File: projects/lib-project/src/lib/dock/view-module/layout/layout.component.ts [hunk 3]
```diff
@@ -6,5 +6,25 @@ import { Component } from '@angular/core';
...
 export class LayoutComponent {
+  backButton : boolean = true;
+  subHeader : any;
+  headerData : any = {};
+  selctedCardItem : any;
+  titleObj = {
+    "title" : "Project name"
+  }
 
+  public sidenavData = [
```
Comment (ramkumar-ns-dev) [inline projects/lib-project/src/lib/dock/view-module/layout/layout.component.ts]: please use with API calls

File: projects/self-creation-portal/src/app/components/app-main-view/app-main-view.component.ts [hunk 4]
```diff
@@ -40,8 +40,8 @@ export class AppMainViewComponent {
   }
 
   getnavData(){
-    this.formService.getForm(SIDE_NAV_DATA).subscribe((form) =>{
-      this.sidenavData = form
+     this.formService.getForm(SIDE_NAV_DATA).subscribe((form) =>{
+      this.sidenavData = form?.result?.data?.fields?.controls
```
Comment (ramkumar-ns-dev) [inline projects/self-creation-portal/src/app/components/app-main-view/app-main-view.component.ts:44]: do we need these many safe operators ?

**Review by ramkumar-pacewisdom — CHANGES_REQUESTED** [review]

**Review by ramkumar-pacewisdom — APPROVED** [review]

**Pull request merged**